### PR TITLE
[Merged by Bors] - update for wgpu 0.8

### PR DESCRIPTION
**Pull request by mockersf**
Changes to get Bevy to compile with wgpu master.

With this, on a Mac:
* 2d examples look fine
* ~~3d examples crash with an error specific to metal about a compilation error~~
* 3d examples work fine after enabling feature `wgpu/cross`

Feature `wgpu/cross` seems to be needed only on some platforms, not sure how to know which. It was introduced in https://github.com/gfx-rs/wgpu-rs/pull/826

### CODE REVIEW

**Comment by mockersf** [comment]
feature `wgpu/cross` can be removed after https://github.com/gfx-rs/naga/pull/693 has been released 🎉

**Comment by ColdIce1605** [comment]
wgpu 0.8 has been released 🎉

**Comment by mockersf** [comment]
updated to 0.8!

examples fails to run with
```
wgpu error: Validation Error

Caused by:
    In Device::create_bind_group
    sampler binding 3 expects filtering = false, but given a sampler with filtering = true
```

**Comment by mockersf** [comment]
fixed above issue by setting all sampler bindings to filtering

File: crates/bevy_wgpu/src/renderer/wgpu_render_context.rs [hunk 1]
```diff
@@ -246,8 +246,8 @@ fn create_wgpu_color_attachment_descriptor<'a>(
     global_render_resource_bindings: &RenderResourceBindings,
     refs: &WgpuResourceRefs<'a>,
     color_attachment_descriptor: &RenderPassColorAttachmentDescriptor,
```
Comment (cart) [inline crates/bevy_wgpu/src/renderer/wgpu_render_context.rs:248]: I think we should probably rename this type for parity with wgpu.

File: crates/bevy_wgpu/src/renderer/wgpu_render_context.rs [hunk 2]
```diff
@@ -269,15 +269,15 @@ fn create_wgpu_depth_stencil_attachment_descriptor<'a>(
     global_render_resource_bindings: &RenderResourceBindings,
     refs: &WgpuResourceRefs<'a>,
     depth_stencil_attachment_descriptor: &RenderPassDepthStencilAttachmentDescriptor,
```
Comment (cart) [inline crates/bevy_wgpu/src/renderer/wgpu_render_context.rs:271]: same here

**Comment by cart** [comment]
I think in general we should probably just rename our types to match wgpu. Mirroring makes it easier for concepts to transfer.

**Comment by mtsr** [comment]
There's a whole bunch more validation in wgpu-0.8, it would be really nice to upgrade just for that alone.

**Comment by mockersf** [comment]
I think I renamed everything following the changes in `wgpu`

File: crates/bevy_wgpu/src/wgpu_type_converter.rs [hunk 3]
```diff
@@ -472,12 +471,12 @@ impl WgpuFrom<FrontFace> for wgpu::FrontFace {
     }
 }
 
-impl WgpuFrom<CullMode> for wgpu::CullMode {
+impl WgpuFrom<CullMode> for Option<wgpu::Face> {
```
Comment (cart) [inline crates/bevy_wgpu/src/wgpu_type_converter.rs:474]: I think this is the only one you missed.

**Pull request closed without merging**